src/ui/client/src/components/Content.js

```diff
@@ -28,32 +28,24 @@ class Content extends React.Component {
     this.loadDestinationData();
   }
 
-  loadDestinationData = async e => {
-    if (e) e.preventDefault();
-
-    const response = await fetch(
-      "/api/v1/destinations/" + this.props.state.suggestion.city + "/" + this.props.state.suggestion.country
-    );
-
-    var data = await response.json();
-
-    console.log(data);
-
-    this.setState({ currentDestination: data });
-  };
+  async componentDidUpdate() {
+    if (this.state.currentDestination.city !== this.props.state.suggestion.city || this.state.currentDestination.country !== this.props.state.suggestion.country) {
+      this.loadDestinationData();
+    }
+  }
 
-  getDestinationData = async (e, { suggestion }) => {
+  loadDestinationData = async e => {
     if (e) e.preventDefault();
 
-    const response = await fetch(
-      "/api/v1/destinations/" + suggestion.city + "/" + suggestion.country
-    );
-
-    var data = await response.json();
+    if (this.props.state.suggestion.city && this.props.state.suggestion.country) {
+      const response = await fetch(
+        "/api/v1/destinations/" + this.props.state.suggestion.city + "/" + this.props.state.suggestion.country
+      );
```

Comment:
Maybe we should decomp the the suggestion?
and use fancy strings
``` js
const { city, country } = this.props.state.suggestion
if (city && country) {
  const response = await fetch(`/api/v1/destinations/${city}/${country}`);
  // ...
}
```

Reply:
Done